### DISCUSSION TITLE
p:sticky to bottom

### DISCUSSION THREAD

**Discussion by matthijs-ll** [primefaces/community#304] [PrimeFaces] [answered]
I'm looking to use sticky to keep the submit button of a long form in view, even when not at the end of the form. I can however find no attributes or examples for this.

**Comment by jepsar** [comment]
Sticky is not made for that scenario. Can't you just use CSS to position it relative to the viewport?

**Reply by matthijs-ll** [reply]
No, it needs to be on its relative position at the bottom of the form when scrolled down - like p:sticky does on top.

CSS even has a widely browser supported "Sticky" display attribute value, combined with top/bottom positioning. **However**, it has no "Stuck state" the way the PrimeFaces sticky functionality has. So there is no way to customize styling when the element is actually stuck to the top or bottom, which is a common feature for stuck items (compact page headers for example).

What we obviously do not want is to introduce separate JS (which you then need to maintain) when all the other UI JS is available from primefaces.

**Reply by melloware** [reply, marked as answer]
@matthijs-ll you can probably steal the code for this: https://www.primefaces.org/showcase/ui/misc/scrollTop.xhtml  see how it makes the button appear in the bottom right corner as you scroll.  Its not much code.
